### PR TITLE
Sema/SIL: NSError has no special powers without ObjC interop.

### PR DESCRIPTION
**Pull request by jckarter**
Explanation: The compiler's NSError bridging code was not sufficiently conditionalized on ObjC interop, and would attempt to toll-free-bridge NSErrors to Error existentials on platforms without ObjC interop.

Scope: Breaks any attempt to work with `corelibs-foundation`, which throws `NSError`s. Catching an `NSError` effectively always triggers a crash.

Issue: SR-585, rdar://problem/30766599

Risk: Low

Testing: Swift CI, manual testing

### CODE REVIEW

**Comment by jckarter** [comment]
@swift-ci Please test

**Comment by jckarter** [comment]
@DougGregor mind reviewing this for inclusion in 3.1?

File: lib/Sema/CSSimplify.cpp [hunk 1]
```diff
@@ -3327,6 +3327,10 @@ ConstraintSystem::simplifyBridgingConstraint(Type type1,
                                              Type type2,
                                              TypeMatchOptions flags,
                                              ConstraintLocatorBuilder locator) {
+  // There's no bridging without ObjC interop.
+  if (!TC.Context.LangOpts.EnableObjCInterop)
```
Comment (DougGregor) [inline lib/Sema/CSSimplify.cpp]: This could be an `assert()`. We should never generate "bridging" constraints unless bridging is enabled.

**Review by DougGregor — APPROVED** [review]

File: lib/Sema/CSSimplify.cpp (hunk 1 above)
Comment (jckarter) [inline lib/Sema/CSSimplify.cpp]: In practice, we do. Should this check be moved up to the places that introduce bridging constraints?

Comment (jckarter) [inline lib/Sema/CSSimplify.cpp]: That might be a good cleanup for master, but I don't think it makes sense to take the risk in 3.1.

**Pull request merged**